radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/TransactionsHandler.java

```diff
@@ -0,0 +1,209 @@
+		var tokenResourceMetadata = (TokenResourceMetadata) substate;
+		return tokenResourceMetadata.getSymbol();
+	}
+
+	private CommittedTransaction construct(Txn txn, RecoverableProcessedTxn recoveryInfo, AccumulatorState accumulatorState) {
+		var parser = radixEngineProvider.get().getParser();
+		ParsedTxn parsedTxn;
+		try {
+			parsedTxn = parser.parse(txn);
+		} catch (TxnParseException e) {
+			throw new IllegalStateException("Could not parse already committed transaction", e);
+		}
+
+		var committedTransaction = new CommittedTransaction();
+		recoveryInfo.recoverStateUpdates(parsedTxn).stream()
+			.map(stateUpdateGroup -> {
+				var operationGroup = new OperationGroup();
+				stateUpdateGroup.stream()
+					.map(stateUpdate -> {
+						var substateOperation = stateUpdate.recover(radixEngineProvider);
+						return coreModelMapper.operation(
+							substateOperation.getSubstate(),
+							substateOperation.getSubstateId(),
+							substateOperation.isBootUp(),
+							this::symbol
+						);
+					})
+					.forEach(operationGroup::addOperationsItem);
+				return operationGroup;
+			})
+			.forEach(committedTransaction::addOperationGroupsItem);
+
+		var signedBy = parsedTxn.getPayloadHashAndSig()
+			.map(hashAndSig -> {
+				var hash = hashAndSig.getFirst();
+				var sig = hashAndSig.getSecond();
+				return ECPublicKey.recoverFrom(hash, sig)
+					.orElseThrow(() -> new IllegalStateException("Invalid signature on already committed transaction"));
+			});
+		var transactionIdentifier = coreModelMapper.transactionIdentifier(txn.getId());
+
+		return committedTransaction
+			.committedStateIdentifier(coreModelMapper.stateIdentifier(accumulatorState))
+			.metadata(new CommittedTransactionMetadata()
+				.fee(coreModelMapper.nativeTokenAmount(parsedTxn.getFeePaid()))
+				.message(parsedTxn.getMsg().map(Bytes::toHexString).orElse(null))
+				.size(txn.getPayload().length)
+				.hex(Bytes.toHexString(txn.getPayload()))
+				.signedBy(signedBy.map(coreModelMapper::publicKey).orElse(null))
+			)
+			.transactionIdentifier(transactionIdentifier);
+	}
+
+	@Override
+	public CommittedTransactionsResponse handleRequest(CommittedTransactionsRequest request) throws CoreApiException {
+		coreModelMapper.verifyNetwork(request.getNetworkIdentifier());
+
+		var stateIdentifier = coreModelMapper.partialStateIdentifier(request.getStateIdentifier());
+		long stateVersion = stateIdentifier.getFirst();
+		var accumulator = stateIdentifier.getSecond();
+		var currentAccumulator = txnStore.getAccumulator(stateVersion)
+			.orElseThrow(() -> CoreApiException.notFound(coreModelMapper.notFoundErrorDetails(request.getStateIdentifier())));
+		if (accumulator != null) {
+			var matchesInput = accumulator.equals(currentAccumulator);
+			if (!matchesInput) {
+				throw CoreApiException.notFound(coreModelMapper.notFoundErrorDetails(request.getStateIdentifier()));
+			}
+		}
+
+		var limit = coreModelMapper.limit(request.getLimit());
```

Comment:
Not sure, I feel like this is constrained by size of the machine and depends on how much memory the node runner gives his node. Adding a cap seems like it would just be an artificial cap.